tests/components/webostv/test_media_player.py

```diff
@@ -1,56 +1,39 @@
 """The tests for the LG webOS media player platform."""
-import unittest
-from unittest import mock
-
-from homeassistant.components.webostv import media_player as webostv
-
-
-class FakeLgWebOSDevice(webostv.LgWebOSDevice):
-    """A fake device without the client setup required for the real one."""
-
-    def __init__(self, *args, **kwargs):
-        """Initialise parameters needed for tests with fake values."""
-        self._source_list = {}
-        self._client = mock.MagicMock()
-        self._name = "fake_device"
-        self._current_source = None
-
-
-class TestLgWebOSDevice(unittest.TestCase):
-    """Test the LgWebOSDevice class."""
-
-    def setUp(self):
-        """Configure a fake device for each test."""
-        self.device = FakeLgWebOSDevice()
-
-    def test_select_source_with_empty_source_list(self):
-        """Ensure we don't call client methods when we don't have sources."""
-        self.device.select_source("nonexistent")
-        assert 0 == self.device._client.launch_app.call_count
-        assert 0 == self.device._client.set_input.call_count
-
-    def test_select_source_with_titled_entry(self):
-        """Test that a titled source is treated as an app."""
-        self.device._source_list = {
-            "existent": {"id": "existent_id", "title": "existent_title"}
-        }
+from unittest.mock import patch
+
+from homeassistant.components import media_player
+from homeassistant.components.media_player.const import (
+    ATTR_INPUT_SOURCE,
+    SERVICE_SELECT_SOURCE,
+)
+from homeassistant.components.webostv import DOMAIN
+from homeassistant.const import ATTR_ENTITY_ID, CONF_HOST, CONF_NAME
+from homeassistant.setup import async_setup_component
+
+
+async def test_select_source_with_empty_source_list(hass):
+    """Test with dummy source."""
+    with patch(
+        "homeassistant.components.webostv.WebOsClient", spec=True,
```

Comment:
I suggest we move this to a pytest fixture function and yield the `return_value` of this patched client class.

Then we can outdent the body of all of the tests that need this patch. We can also easily access the mocked client and assert things about it.

```py
@pytest.fixture(name="client")
def client_fixture():
    with patch(...) as mock_client_class:
        yield mock_client_class.return_value


async def test_select_source_with_empty_source_list(hass, client):
```

Reply:
Any suggestions for working around the lack of AsyncMock in python 3.7?

Reply:
We have `asynctest` library available as a core requirement so we can use `asynctest.patch` and `asynctest.CoroutineMock`.